pkg/auth/interceptor.go

```diff
@@ -26,23 +27,48 @@ const (
 	projectHeader = "Openai-Project"
 )
 
+// Config is the configuration for an Interceptor.
+type Config struct {
+	RBACServerAddr string
+
+	// AccessResource is the static resource name to access. This value or GetAccessResource functions must be set.
+	AccessResource string
+	// GetAccessResourceForGRPCRequest is a function to get the resource name from a gRPC method.
+	GetAccessResourceForGRPCRequest func(fullMethod string) string
```

Comment:
Question/confirmation: Will we add only set `GetAccessResourceForGRPCRequest`? Just curious as the code allows it and it's useful when the server only implements gRPC handlers.

Reply:
Yes, this PR supports grpc and http, but I plan to use only grpc for now.

Reply:
Sounds good!